### PR TITLE
enh(dashboard): remove version column from dashboard_widgets table

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

this PR intends to  remove version column from dashboard_widgets table

**Fixes** # MON-24909

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php [hunk 1]
```diff
@@ -82,10 +82,11 @@ public function getWidgetsVersion(string $webVersion): array
             }
         }
 
-        $dashboardWidgetsRequest = $this->translateDbName('SELECT `name`, `version` FROM `:db`.dashboard_widgets');
+        $dashboardWidgetsRequest = $this->translateDbName('SELECT `name` FROM `:db`.dashboard_widgets');
+        $version = $this->getWebVersion();
         if (($statement = $this->db->query($dashboardWidgetsRequest)) !== false) {
             while ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
-                $versions[(string) $result['name']] = (string) $result['version'];
+                $versions[(string) $result['name']] = $version;
```
Comment (wtermellil) [inline centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php:89]: According `getWebVersion` `$version` can be null, won't that be a problem ?

**Review by arnapou — APPROVED** [review]

File: centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php (hunk 1 above)
Comment (jeremyjaouen) [reply centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php:89]: Nope, that was already handle like this for other widgets or web version

Comment (arnapou) [reply centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php:89]: Nice spot :+1:

Comment (wtermellil) [reply centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php:89]: @jeremyjaouen For the future we should rework `getWebVersion()' and throw an exception if no version is available

**Review by wtermellil — APPROVED** [review]

**Pull request merged**